lib/ash_json_api/serializer.ex

```diff
@@ -399,7 +399,7 @@ defmodule AshJsonApi.Serializer do
 
   defp add_one_record_self_link(links, request, %resource{} = record) do
     resource
-    |> AshJsonApi.Resource.route(%{action: :get, primary?: true})
+    |> AshJsonApi.Resource.route(%{method: :get, primary?: true})
```

Comment:
I think we should do `route(%{type: :get})` in this case, actually since that will prevent us from using index requests.

Reply:
Sure will do 👍